app/src/main/java/zegreanu/cristi/geoquiz/QuizActivity.kt

```diff
@@ -1,18 +1,26 @@
 package zegreanu.cristi.geoquiz
 
+import android.app.Activity
+import android.content.Context
+import android.content.Intent
 import android.os.Bundle
 import android.util.Log
 import android.widget.Button
 import android.widget.TextView
 import android.widget.Toast
 import androidx.appcompat.app.AppCompatActivity
+import kotlin.properties.Delegates
 
 class QuizActivity : AppCompatActivity() {
     private val TAG = "QuizActivity"
     private val KEY_INDEX = "index"
+    private val EXTRA_ANSWER_IS_TRUE = "zegreanu.cristi.geoquiz.answer_is_true"
+    private val EXTRA_ANSWER_SHOWN = "zegreanu.cristi.geoquiz.answer_shown"
```

Comment:
Ideally this should be defined only on the destination activity and have it public - as in someway it's part of the "callback" (onActivityResult) public API.

Reply:
You forgot to move EXTRA_ANSWER_SHOWN